### PR TITLE
Fixed xonfig raising error when xonsh is not installed from source

### PR DESCRIPTION
**Pull request by mitnk**
Found xonfig is broken when I was trying 0.5.0 in a clean env..

Sorry @scopatz, maybe you could do a 0.5.1 release for it soon?

### CODE REVIEW

File: tests/test_platform.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def test_githash_value_error(monkeypatch):
+    @contextmanager
+    def mocked_open(*args):
+        yield MagicMock(read=lambda: 'abc123')
+    monkeypatch.setattr(builtins, 'open', mocked_open)
+    sha, date_ = xp.githash()
+    assert date_ is None
+    assert sha is None
```
Comment (mitnk) [inline tests/test_platform.py]: Tried to add a normal case, but haven't found a way to mock up `functools.lru_cache`. So only bad case added.

**Comment by gforsyth** [comment]
Thanks, @mitnk! I'll try to push a microrelease but I'm on _SLOW_ internet and might have to have one of the other core devs do it

**Comment by mitnk** [comment]
I think it's okay, not a urgent one :) And better this release could include this one https://github.com/xonsh/xonsh/pull/2028

**Comment by gforsyth** [comment]
well, too late, but we can always do another one

**Comment by gforsyth** [comment]
and I, for one, would rather `xonfig` were working after a major release since it's the first thing we ask of users reporting issues

**Comment by mitnk** [comment]
haha. agreed. that one is really minor compare to xonfig. Thank you!

**Pull request merged**